src/nvim/charset.c

```diff
@@ -424,7 +424,7 @@ char_u *str_foldcase(char_u *str, int orglen, char_u *buf, int buflen)
   int i;
   int len = orglen;
 
-#define GA_CHAR(i) ((char_u *)ga.ga_data)[i]
+#define GA_CHAR(i) ((char *)ga.ga_data)[i]
```

Comment:
hmm, is a bit weird to have this different type than `buf[i]` (though this entire function is a convoluted mess. all these macros can be eliminated by just setting `buf = ga.ga_data` , though that would be another PR)